clang/test/Sema/main-no-return-c89.c

```diff
@@ -0,0 +1,6 @@
+/* RUN: %clang_cc1 -std=c89 -fsyntax-only -verify -pedantic -Wno-strict-prototypes %s
+ */
+
+int main() {
```

Comment:
You should also have a test where `main` does return a value to show that we silence the diagnostic, and another test along the lines of:
```
int main() {
  if (1)
    exit(1);
}
```
which should also silence the diagnostic.